src/app/settings/admin/constraint-templates/component.ts

```diff
@@ -0,0 +1,154 @@
+// Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, OnDestroy, OnInit, ViewChild} from '@angular/core';
+import {MatDialog, MatDialogConfig} from '@angular/material/dialog';
+import {MatPaginator} from '@angular/material/paginator';
+import {MatSort} from '@angular/material/sort';
+import {MatTableDataSource} from '@angular/material/table';
+import {ApiService} from '@core/services/api/service';
+import {NotificationService} from '@core/services/notification/service';
+import {UserService} from '@core/services/user/service';
+import {ConfirmationDialogComponent} from '@shared/components/confirmation-dialog/confirmation-dialog.component';
+import {ConstraintTemplate} from '@shared/entity/opa';
+import * as _ from 'lodash';
+import {Subject} from 'rxjs';
+import {takeUntil, filter, take, switchMap} from 'rxjs/operators';
+import {ConstraintTemplatesDataDialogComponent} from './constraint-templates-data-dialog/component';
+
+@Component({
+  selector: 'km-constraint-templates',
+  templateUrl: './template.html',
+})
+export class ConstraintTemplatesComponent implements OnInit, OnDestroy {
```

Comment:
Decided to rename the component, as now I think it is telling more about the content.